learning_resources/views.py

```diff
@@ -179,4 +200,86 @@ def get_queryset(self):
         Returns:
             QuerySet of LearningResource objects that are Programs
         """
-        return self._get_base_queryset(resource_type=LearningResourceType.program.value)
+        return self._get_base_queryset(
+            resource_type=LearningResourceType.program.value
+        ).filter(published=True)
+
+
+class LearningPathViewSet(LearningResourceViewSet, viewsets.ModelViewSet):
+    """
+    Viewset for LearningPaths
+    """
+
+    serializer_class = LearningPathResourceSerializer
+    permission_classes = (permissions.HasLearningPathPermissions,)
+
+    def get_queryset(self):
+        """
+        Generate a QuerySet for fetching valid Programs
+
+        Returns:
+            QuerySet of LearningResource objects that are Programs
+        """
+        queryset = self._get_base_queryset(
+            resource_type=LearningResourceType.learning_path.value,
+        )
+        if not (is_learning_path_editor(self.request) or is_admin_user(self.request)):
+            queryset = queryset.filter(published=True)
+        return queryset
+
+    def create(self, request, *args, **kwargs):
+        request.data["readable_id"] = uuid4().hex
+        request.data["resource_type"] = LearningResourceType.learning_path.value
+        return super().create(request, *args, **kwargs)
+
+    def perform_destroy(self, instance):
+        # deindex_staff_list(instance.learning_resource)
+        instance.delete()
+
+
+class ResourceListItemsViewSet(NestedViewSetMixin, viewsets.ModelViewSet):
+    """
+    Viewset for LearningResource related resources
+    """
+
+    permission_classes = (AnonymousAccessReadonlyPermission,)
+    serializer_class = LearningResourceChildSerializer
+    queryset = (
+        LearningResourceRelationship.objects.select_related("child")
+        .prefetch_related(
+            "child__runs",
+            "child__runs__instructors",
+        )
+        .order_by("position")
+    )
+    pagination_class = DefaultPagination
+
+    def get_parent_id(self):
+        """Get the parent id for the nesteed view request"""
+        return self.get_parents_query_dict()["parent_id"]
+
+
+class LearningPathItemsViewSet(ResourceListItemsViewSet):
+    """
+    Viewset for LearningPath related resources
+    """
+
+    serializer_class = LearningPathRelationshipSerializer
+    permission_classes = (permissions.HasLearningPathItemPermissions,)
+
+    def create(self, request, *args, **kwargs):
+        request.data["parent"] = self.get_parent_id()
+        return super().create(request, *args, **kwargs)
```

Comment:
I was curious how `drf-spectacular` would deal with views modifying request data like this, so I looked into that a bit. Without any extra work, the OpenAPI spec thinks `parent` is required.

The only way I could find to model this with `drf-spectacular` is something along the lines of:
```python
    @extend_schema(
            request=extend_schema_serializer(
                exclude_fields=["parent"]
            )(LearningPathRelationshipSerializer)
    )
    def create(self, request, *args, **kwargs):
```

I'm not suggesting we add that, just writing it down for future. It's a little verbose and if the OpenAPI Spec thinks `requestBody.parent` is required, it's not really problematic since whoever is calling the API must have it already for the path param.

Reply:
I wonder if there is a way to hide CRUD API specs altogether and just show the read-only API specs. Seems like CRUD requests should only happen via our own frontend (and probably need to, due to Django CSRF).

Reply:
@mbertrand I'm sure we could hide the CRUD (except R) API specs.

IMO, the CUD spec is somewhat useful internally since we generate the API client on our frontend. But I don't want to overstate that... there's still manual work: We generate the API client automatically, but in order to use it, we have to write some react hooks, and that portion is manual anyway.